lib/asciidoctor/converter/semantic-html5.rb

```diff
@@ -86,6 +91,35 @@ def generate_authors node
     end
   end
 
+  def generate_revision node
+    return unless (node.attr? 'revnumber') || (node.attr? 'revdate') || (node.attr? 'revremark')
+
+    revision_date = if (revdate = node.attr 'revdate')
+      date = ::Date._parse revdate
+      date_text = "#{date[:year]}-#{date[:mon].to_s.rjust 2, '0'}-#{date[:mday].to_s.rjust 2, '0'}"
+
+      %(<time datetime="#{date_text}">#{revdate}</time>)
+    else
+      ''
+    end
+    %(<table class="revision">
+<thead>
+<tr>
+<th>Version</th>
+<th>Date</th>
+<th>Remark</th>
+</tr>
+</thead>
+<tbody>
+<tr>
+<td data-title="#{node.attr 'version-label', ''}">#{node.attr 'revnumber', ''}</td>
```

Comment:
Perhaps the version-label should control the column header instead of being added per cell. It also seems a bit redundant to add data-title to each cell. wdyt?

Reply:
I'm using `data-title` to preprend the version label using CSS:

```css
table.revision tr > td:first-child::before {
  content: attr(data-title) ' '
}
```

It would be nice if we could get the value of the header for a given cell in CSS but as far as I know that's not possible 😞 

> It also seems a bit redundant to add data-title to each cell. wdyt?

That's true but I guess that's also true if we want to apply a custom style on the first cell in a table. In this case, we would need to add a class on each cell of every row.

Alternatively, we could generate dynamic CSS:

```ruby
css = %(table.revision tr > td:first-child::before {
  content: #{node.attr 'version-label', ''} ' '
})
```

But we should be cautious because if we do that then we won't have a fully static CSS...

> Perhaps the version-label should control the column header instead of being added per cell.

Do you think we should also use `#{node.attr 'version-label', ''}` in the table header? If so, I believe that "Remark" and "Date" should also be configurable (for i18n).

Reply:
Thanks for explaining. Now I understand.

I agree that putting the version label in a data attribute makes styling it more flexible (which is a core goal of this converter). I think I'd rather see it as a span element, though, rather than on the cell.

```
<td><span data-title="#{node.attr 'version-label'}">#{node.attr 'revnumber', ''}</span></td>
```

I don't think the other cells should have data-title attributes since these are already provided by the table header.

> Do you think we should also use #{node.attr 'version-label', ''} in the table header? If so, I believe that "Remark" and "Date" should also be configurable (for i18n).

Perhaps we will want that. But it should be a separate issue since it entails adding new attributes...names which we then have to support.

Reply:
Btw, you don't need the fallback string when resolving the version-label attribute:

```
node.attr 'version-label'
```

That's only needed if you want to call a method on the result (though even then, Ruby 2.5 has a safe navigation operator).

Reply:
> Btw, you don't need the fallback string when resolving the version-label attribute:

I didn't know that the string representation of `nil` is `''`, I will remove it 😄 

> Perhaps we will want that. But it should be a separate issue since it entails adding new attributes...names which we then have to support.

👍🏻 

> I agree that putting the version label in a data attribute makes styling it more flexible (which is a core goal of this converter). I think I'd rather see it as a span element, though, rather than on the cell.

Any rationale behind this decision?

Reply:
> I don't think the other cells should have data-title attributes since these are already provided by the table header.

Unless, you are hiding the table header.
It gives the opportunity to tweak the design:

```css
thead {
  display: none;
}
tbody > tr {
  display: flex;
  flex-direction: column;
}
tbody > tr > td {
  display: flex;
  gap: 5px;
}
tbody > tr > td:nth-child(1)::before,
tbody > tr > td:nth-child(2)::before {
  content: attr(data-title) ' ';
  display: block;
  font-weight: bold;
}
```

![image](https://user-images.githubusercontent.com/333276/115905297-318ee580-a466-11eb-927c-6dc57bde62d6.png)

Reply:
> Any rationale behind this decision?

Because "before" on a table cell fundamentally changes the DOM structure of the HTML. It essentially adds another element next to the table cell, which is really not semantic at all.

> Unless, you are hiding the table header.
It gives the opportunity to tweak the design:

That seems like some pretty drastic CSS hacking. I would think it would be better in that case to extend the converter. I feel like we are pushing the semantics too far here.

Reply:
> I feel like we are pushing the semantics too far here.

The `data-title` attribute is all about CSS styling. Semantically, revisions are well structured in a table.

> Because "before" on a table cell fundamentally changes the DOM structure of the HTML. It essentially adds another element next to the table cell, which is really not semantic at all.

You mean inside the table cell?
I'm struggling to understand what value provides the `<span>` element:

##### before

```html
<td>
  ::before
  revnumber
</td>
```

##### after

```html
<td>
  <span>
    ::before
    revnumber
  </span>
</td>
```

On one hand, we add a pseudo-element in a `<td>` element and on the other hand we add a pseudo-element in a `<span>` element (inside a `<td>` element).

I'm voting in favor of leaving this for later and removing the `data-title` attribute for now.
We are making adjustments and speculating on how to style a component rather than focusing on establishing a well structured HTML.

I think we've agreed that the revision

Reply:
I did some more testing and realized I was incorrect. The pseudo-element does go *inside* the table cell. So it's fine to put data-title on the td itself.

I see your point about adding data-title to all the cells to accommodate advanced layouts. So I'm willing to go with and see how it plays out.